lib/http_requests.js

```diff
@@ -24,17 +24,24 @@ exports.fetchResource = function fetchResource(t, callback) {
     // The step already has an attached result document, so all is fine and we
     // can call the callback immediately
     log.debug('resource for next step has already been fetched, using ' +
-        'embedded');
-    return process.nextTick(function() {
-      callback(null, t);
-    });
+      'embedded');
+    
+    if (process && Object.hasOwnProperty.call(process, 'nextTick')) {
+      return process.nextTick(function() {
+        callback(null, t);
+      });
+    }
+    return callback(null, t);
   } else {
-    return process.nextTick(function() {
-      var error = createError('Can not process step.',
-        errors.InvalidStateError);
-      error.step = t.step;
-      callback(error, t);
-    });
+    var error = createError('Can not process step.', errors.InvalidStateError);
+    error.step = t.step;
+
+    if (process && Object.hasOwnProperty.call(process, 'nextTick')) {
```

Comment:
Instead of doing this check every time we want to use `nextTick`, we could do it at the top of the module once and save it to a boolean variable.

This also applies to other places where you introduced this check.

Reply:
I moved it to the top of the module for every place in commit 8b6ed6d